### ISSUE TITLE
[ENH] change loss dtype from `nn.Module` to `MultiHorizonMetric`.

### ISSUE THREAD

**Issue by PranavBhatP** [sktime/pytorch-forecasting#1844]
**Is your feature request related to a problem? Please describe.**
As a part of `v2`, the dtype of `loss` under the `BaseModel` has been made `nn.Module`. This can be replaced with the existing `MultiHorizonMetric`. This implementation is more robust and doesn't require us to implement new loss functions.

If there is a specific reason for this change, please do let me know.

**Comment by PranavBhatP** [comment]
FYI @phoeenniixx

**Comment by phoeenniixx** [comment]
Yes this can be a good idea! While creating the base class I just tried to get the basic implementation, and used the metrics I have experience with, but after looking at your comment I noticed even v1's `BaseModel` used `Metric`, so I think we should do that too...
(`MultiHorizonMetric` inherits from `Metric`, so maybe we should use `Metric`?)
